lib/admin/ui/pages/edit_module_visibility/edit_module_visibility.dart

```diff
@@ -0,0 +1,130 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:heroicons/heroicons.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:myecl/admin/class/module_visibility.dart';
+import 'package:myecl/admin/providers/all_groups_list_provider.dart';
+import 'package:myecl/admin/providers/all_module_visibilities_list_provider.dart';
+import 'package:myecl/admin/ui/admin.dart';
+import 'package:myecl/admin/providers/module_visibilities_list_provider.dart';
+
+class EditModulesVisibilityPage extends HookConsumerWidget {
+  const EditModulesVisibilityPage({super.key});
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final modulesNotifier = ref.watch(moduleVisibilitiesListProvider.notifier);
+
+    final modules = ref.watch(allModuleVisibilitiesList);
+    final groups = ref.watch(allGroupList);
+
+    final isOpen = useState<List<bool>>([]);
```

Comment:
Maybe you can go for another solution, by adding a isOpened parameter to your ModuleVisibilities class.
Thus, you can control each moduleVisibilities individually through the moduleVisibilities provider and not using this array.

Your openModule provider function can be an update with no API call (done with a () async => true function as API call).

It will be cleaner and easier to understand and maintain